packages/css-engine/src/core/rules.ts

```diff
@@ -65,11 +65,22 @@ export type MediaRuleOptions = {
 };
 
 export class MediaRule {
-  #options: MediaRuleOptions;
+  // Sort media rules by minWidth.
+  // Needed to ensure that more specific media rules are inserted after less specific ones.
+  // So that they get a higher specificity.
+  static sort(mediaRules: Iterable<MediaRule>) {
+    return Array.from(mediaRules).sort((ruleA, ruleB) => {
+      return (
```

Comment:
-Infinity - (-Infinity) = NaN
Looks like undefined behavior?

Reply:
oh amazing behavior from javascript, will switch to -Number.MAX_SAFE_INTEGER